File: languages/python/04-expert/modularization-and-build-structure/example/formatting.py
# This helper example focuses on keeping presentation rules separate from calculations and program
# startup.

from __future__ import annotations


# Keep this helper separate so the main example can focus on the larger idea without extra noise.
def render_summary(summary: dict[str, float]) -> str:
    return (
        f"Subtotal: {summary['subtotal']:.2f}\n"
        f"Discount: {summary['discount_value']:.2f}\n"
        f"Tax: {summary['tax_value']:.2f}\n"
        f"Total: {summary['total']:.2f}"
    )

File: languages/go/04-expert/modularization-and-build-structure/example/formatting.go
// This helper example focuses on keeping presentation rules separate from calculations and program startup.

package main

import "fmt"

// Keep this helper separate so the main example can focus on the larger idea without extra noise.
func renderSummary(summary invoiceSummary) string {
	return fmt.Sprintf(
		"Subtotal: %.2f\nDiscount: %.2f\nTax: %.2f\nTotal: %.2f",
		summary.subtotal,
		summary.discountValue,
		summary.taxValue,
		summary.total,
	)
}
